File: iosApp/iosApp/Views/Components/BasicText.swift
//
//  BasicText.swift
//  iosApp
//
//  Created by Jan Cortiel on 07.02.23.
//  Copyright © 2023 Ludwig Boltzmann Institute for
//  Digital Health and Prevention - A research institute
//  of the Ludwig Boltzmann Gesellschaft,
//  Oesterreichische Vereinigung zur Foerderung
//  der wissenschaftlichen Forschung
//  Licensed under the Apache 2.0 license (see https://www.apache.org/licenses/LICENSE-2.0).
//

import SwiftUI

struct BasicText: View {
    var text: String
    var color: Color = Color.more.primary
    var font: Font = Font.body
    var lineLimit: Int? = nil
    var textAlign: TextAlignment = .leading
    var body: some View {
        Text(LocalizedStringKey(text))
            .foregroundColor(color)
            .multilineTextAlignment(textAlign)
            .fixedSize(horizontal: false, vertical: true)
            .font(font)
            .lineLimit(lineLimit)
            .truncationMode(.tail)
    }
}

struct BasicText_Previews: PreviewProvider {
    static var previews: some View {
        BasicText(text: "Hello World")
    }
}

File: androidApp/src/main/java/io/redlink/more/app/android/shared_composables/BasicText.kt
/*
 * Copyright LBI-DHP and/or licensed to LBI-DHP under one or more
 * contributor license agreements (LBI-DHP: Ludwig Boltzmann Institute
 * for Digital Health and Prevention -- A research institute of the
 * Ludwig Boltzmann Gesellschaft, Österreichische Vereinigung zur
 * Förderung der wissenschaftlichen Forschung).
 * Licensed under the Apache 2.0 license with Commons Clause
 * (see https://www.apache.org/licenses/LICENSE-2.0 and
 * https://commonsclause.com/).
 */
package io.redlink.more.app.android.shared_composables

import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.TextUnit
import io.redlink.more.app.android.ui.theme.MoreColors

@Composable
fun BasicText(
    text: String,
    modifier: Modifier = Modifier,
    color: Color = MoreColors.Primary,
    textAlign: TextAlign? = null,
    fontSize: TextUnit = TextUnit.Unspecified,
) {
    Text(
        text = text,
        color = color,
        textAlign = textAlign,
        fontSize = fontSize,
        modifier = modifier
    )
}